astropy/units/physical.py

```diff
@@ -1,133 +1,461 @@
+        new_unit = self._unit / other_unit
+        return new_unit.physical_type
+
+    def __rtruediv__(self, other):
+        other_unit = _identify_unit_from_unit_or_physical_type(other)
+        other_unit = _replace_temperatures_with_kelvin(other_unit)
+        new_unit = other_unit / self._unit
+        return new_unit.physical_type
+
+    def __pow__(self, power):
+        if not isinstance(power, numbers.Real):
+            raise TypeError(f"{power} is not a real number")
+        return (self._unit ** power).physical_type
+
+    def __hash__(self):
+        return hash(self._physical_type_id)
+
+
+def _get_names_in_use_except_from(unit):
+    all_names_in_use = set(_unit_physical_mapping.keys())
+
+    id = unit._get_physical_type_id()
+    if id in _physical_unit_mapping.keys():
+        names_for_this_unit = set(_physical_unit_mapping[id])
+    else:
+        names_for_this_unit = set()
+
+    return all_names_in_use - names_for_this_unit
+
+
 def def_physical_type(unit, name):
     """
-    Adds a new physical unit mapping.
+    Add a mapping between a unit and the corresponding physical type(s).
+
+    If a physical type already exists for a unit, add new physical type
+    names so long as those names do not correspond to other units.
+
+    This function is not intended to be called directly in user code.
 
     Parameters
     ----------
-    unit : `~astropy.units.UnitBase` instance
-        The unit to map from.
+    unit : u.Unit
+        The unit to be represented by the physical type.
+
+    name : `str` or `set` of `str`
+        A `str` representing the name of the physical type of the unit,
+        or a `set` containing strings that represent one or more names
+        of physical types.
 
-    name : str
-        The physical name of the unit.
+    Raises
+    ------
+    ValueError
+        If a physical type name is already in use for another unit, or
+        if attempting to name a unit as ``"unknown"``.
     """
-    r = unit._get_physical_type_id()
-    if r in _physical_unit_mapping:
+
+    name = _standardize_physical_type_names(name)
+
+    if "unknown" in name:
+        raise ValueError("cannot uniquely define an unknown physical type")
+
+    names_for_other_units = _get_names_in_use_except_from(unit)
+
+    names_already_in_use = name & names_for_other_units
+    if names_already_in_use:
         raise ValueError(
-            f"{r!r} ({name!r}) already defined as {_physical_unit_mapping[r]!r}")
-    _physical_unit_mapping[r] = name
-    _unit_physical_mapping[name] = r
+            f"the following physical type names are already in use: "
+            f"{names_already_in_use}."
+        )
+
+    physical_type_id = unit._get_physical_type_id()
+    unit_already_in_use = physical_type_id in _physical_unit_mapping
+    if unit_already_in_use:
+        old_physical_type = _physical_unit_mapping[physical_type_id]
+        name |= set(old_physical_type)
```

Comment:
Are you instantiating a new physical type to ensure they are immutable?

Reply:
I didn't have any particular reason.  I ended up changing it so that it's the same `PhysicalType` instance, only re-initialized with the new names.  Seems like this will be a slight improvement.